daemon/cluster/filters_test.go

```diff
@@ -0,0 +1,327 @@
+package cluster
+
+import (
+	"testing"
+
+	"github.com/docker/docker/api/types/filters"
+)
+
+func TestNewListNodesFilters(t *testing.T) {
+	validFilters := []filters.Args{
+		{
+			fields: map[string]map[string]bool{
+				"name": {"nodeName": true},
+			},
+		},
+		{
+			fields: map[string]map[string]bool{
+				"id": {"7c9009d6720f6de3b492f5": true},
+			},
+		},
+		{
+			fields: map[string]map[string]bool{
+				"label": {"type=test": true, "storage=ssd": true, "memory": true},
+			},
+		},
+		{
+			fields: map[string]map[string]bool{
+				"role": {"manager": true},
+			},
+		},
+		{
+			fields: map[string]map[string]bool{
+				"role": {"manager": true, "worker": true},
+			},
+		},
+		{
+			fields: map[string]map[string]bool{
+				"membership": {"accepted": true},
+			},
+		},
+		{
+			fields: map[string]map[string]bool{
+				"membership": {"accepted": true, "pending": true},
+			},
+		},
+		{
+			fields: map[string]map[string]bool{
+				"name":       {"nodeName": true},
+				"id":         {"7c9009d6720f6de3b492f5": true},
+				"label":      {"type=test": true, "storage=ssd": true, "memory": true},
+				"role":       {"manager": true},
+				"membership": {"accepted": true},
+			},
+		},
+	}
+
+	invalidFilters := []filters.Args{
+		{
+			fields: map[string]map[string]bool{
+				"Name": {"asasa": true},
+			},
+		},
+		{
+			fields: map[string]map[string]bool{
+				"type": {"a": true},
+			},
+		},
+		{
+			fields: map[string]map[string]bool{
+				"role": {"Worker": true},
+			},
+		},
+		{
+			fields: map[string]map[string]bool{
+				"role": {"worker": true, "boss": true},
+			},
+		},
+		{
+			fields: map[string]map[string]bool{
+				"role": {"worker", true, "manager": true, "worker_and_manager": true},
```

Comment:
This test is testing the same as above right ? (an invalid value in the middle of valid ones) — if yes, could be remove 👼